app/apps/route.js

```diff
@@ -13,5 +13,11 @@ export default Ember.Route.extend({
     } else {
       return apps;
     }
+  },
+  redirect: function() {
+    var stack = this.modelFor('stack');
+    if(stack.get('apps.length') === 0) {
+      this.replaceWith('apps.new', this.modelFor('stack'));
+    }
```

Comment:
The `redirect` hook is passed the resolved `model` (the array of apps), so I think you can do:
```
redirect: function(model){
  if (model.get('length') === 0) {
    this.transitionTo('stack.apps.new', this.modelFor('stack');
  }
}
```